### ISSUE TITLE
Refactor apiQuerier and ExecutorAPI

### ISSUE THREAD

**Issue by MrSerth** [openHPI/poseidon#114]
_This issue has been migrated from [GitLab !59](https://gitlab.hpi.de/codeocean/codemoon/poseidon/-/issues/59). See discussion_

## Original User Story

**As a** developer  
**I want to** merge the apiQuerier and ExecutorAPI interfaces  
**In order to** simplify the code.

## Original Acceptance Criteria

- [ ] there is only one interface encapsulating the access to the Nomad API
- [ ] this interface does not longer contain domain knowledge on our use case but should only have to care about posing API queries to nomad
  - [ ] wrapper methods that contain domain knowledge should be moved to the object that is responsible 

## Discussion

In a discussion with @mpass99, we identified various advantages and disadvantages of the current architecture or a merged one:

+ Nomad Client is only used in the apiQuerier (only visible in the `nomad` package)
+ Separation of methods using the Nomad Client (apiQuerier) vs those building on top of these methods (ExecutorAPI)
- apiQuerier has some methods that are being available in other packages due to the apiClient implementing both interfaces
- Some methods (such as `Execute` in the apiQuerier) semantically feel to be misplaced but currently require raw access to the Nomad Client

In summary, we agreed on some refactoring potential to either increase the separation between both parts (by moving those methods from the apiQuerier to the ExecutorAPI that are used outside) or continue with the original merge. As we don't see any urgent need (or another issue that is blocked by this refactoring), we decided to keep this idea on hold for the time being.
